### PR TITLE
DOCS-107 cleanup and new examples #time 1h

### PR DESCRIPTION
**Pull request by ptimmons**
The original docs were a bit tough to follow. I added datamodel examples (for consistency with our reference guide) and renamed some of the config elements to hopefully make the use of the plugin more clear.

### CODE REVIEW

File: docs/plugin_dns_cache.md [hunk 1]
```diff
@@ -3,7 +3,7 @@ title: DNS Cache Plugin
...
+The 128T-dns-cache plugin provides a DNS caching service on your 128T router by forwarding all traffic on a configured `ingress-service` and `tenant` to [dnsmasq](http://www.thekelleys.org.uk/dnsmasq/doc.html). By default it uses existing system configuration for nameserver resolution, but supports optionally configuring custom server addresses. The plugin is implemented as a service function chain (SFC) where the DNS traffic received on an ingress interface is detoured through a Kernel Network Interface (KNI) to be cached by `dnsmasq`. A typical flow of the packets for the plugin is illustrated below:
```
Comment (agrawalkaushik) [inline docs/plugin_dns_cache.md:6]: We could include links to the appropriate documentations for `service function chain (SFC)` and `Kernel Network Interface (KNI)` if you'd like. They can be following:

plugins_intro.md#service-function-chaining
concepts_kni.md

Comment (ptimmons) [reply docs/plugin_dns_cache.md]: Done, thanks.

File: docs/plugin_dns_app_id.md [hunk 2]
```diff
@@ -34,24 +36,37 @@ configure
...
+| custom-apps > description | string | A human-readable description of the custom application. |
+| custom-apps > patterns | regex | A regular expression pattern for matching to DNS requests that the 128T processes. |
+
 ### Built-In Patterns
 
 The plugin contains built in application patterns for Gmail, Google Drive, and Windows Update. The patterns were generated from the following published documents:
 
 | Application    | Document |
 | -------------- | --- |
-| Gmail          | [link](https://support.google.com/a/answer/9497877?hl=en) |
```
Comment (robert-patterson) [inline docs/plugin_dns_app_id.md:43]: Should this space be there?

Comment (MichaelBaj) [reply docs/plugin_dns_app_id.md:43]: The build is failing on docs/plugin_dns_cache/plugins_intro.md#service-function-chaining - which is clearly an invalid link.

Comment (MichaelBaj) [reply docs/plugin_dns_app_id.md:43]: I pushed up a commit to fix the typo in the filename

**Pull request merged**